src/pages/transactions/edit/Edit.tsx

```diff
@@ -0,0 +1,207 @@
+import { request } from 'common/helpers/request';
+import toast from 'react-hot-toast';
+import { useNavigate, useParams } from 'react-router-dom';
+import { route } from 'common/helpers/route';
+
+const Edit = () => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const { id } = useParams<string>();
+  const currencies = useCurrencies();
+  const pages = useTransactionPages();
+  const { documentTitle } = useTitle('edit_transaction');
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [errors, setErrors] = useState<TransactionValidation>();
+  const [transaction, setTransaction] = useState<TransactionInput>(
+    defaultTransactionProperties
+  );
+  const { checkValidation, setValidation } = useTransactionValidation();
+
+  const getTransactionInputObject = (
+    responseDetails: any
+  ): TransactionInput => {
+    const { date, amount, currency_id, bank_integration_id, description } =
+      responseDetails || {};
+    return {
+      base_type:
+        responseDetails?.base_type === 'CREDIT' ? 'deposit' : 'withdrawal',
+      date,
+      amount,
+      currency_id,
+      bank_integration_id,
+      description,
+    };
+  };
+
+  const fetchTransactionDetails = async () => {
+    const response = await request(
+      'GET',
+      endpoint('/api/v1/bank_transactions/:id', { id })
+    );
+    setIsLoading(false);
+    setTransaction(getTransactionInputObject(response?.data?.data));
+  };
+
+  const handleChange = (
+    property: keyof TransactionInput,
+    value: unknown
+  ): void => {
+    setTransaction(
+      (transaction) => transaction && { ...transaction, [property]: value }
+    );
+  };
+
+  const onSave = async (event: React.FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+    setErrors(undefined);
+    const isFormValid = checkValidation(transaction);
+
+    if (!isFormValid && !errors) {
+      setValidation(transaction, setErrors, errors);
+    }
+
+    if (isFormValid) {
+      setIsSaving(true);
+      const toastId = toast.loading(t('processing'));
+      try {
+        await request(
+          'PUT',
+          endpoint('/api/v1/bank_transactions/:id', { id }),
+          {
+            ...transaction,
+            amount: +transaction?.amount,
+            base_type:
+              transaction?.base_type === 'deposit' ? 'CREDIT' : 'DEBIT',
+          }
+        );
+        toast.success(t('updated_transaction'), { id: toastId });
+        setIsSaving(false);
+        navigate(route('/transactions'));
+      } catch (error: any) {
+        setIsSaving(false);
+        console.error(error);
+        if (error.response?.status == 422) {
+          setErrors(error.response.data);
+        }
+        toast.error(t('error_title'), { id: toastId });
+      }
+    }
+  };
+
+  useEffect(() => {
+    fetchTransactionDetails();
```

Comment:
Is there a reason why fetch like this and not using React Query with separate hook for the query?

Reply:
> Is there a reason why fetch like this and not using React Query with separate hook for the query?

changed